vault/activity_log_test.go

```diff
@@ -1640,6 +1642,25 @@ func TestActivityLog_DeleteWorker(t *testing.T) {
 	expectMissingSegment(t, core, ActivityLogPrefix+"directtokens/1111/1")
 }
 
+// checkAPIWarnings ensures there is a warning if switching from enabled -> disabled,
+// and no response otherwise
+func checkAPIWarnings(t *testing.T, originalEnabled, newEnabled bool, resp *logical.Response) {
+	t.Helper()
+
+	expectWarning := originalEnabled == true && newEnabled == false
+
+	if !expectWarning && resp != nil {
```

Comment:
I find switch statements more readable than nested if/elses like this, but YMMV.

Reply:
hmm, how would you set up this particular one as a switch?

Reply:
```
switch {
case !expectWarning && resp != nil: 
	t.Fatalf("got unexpected response: %#v", resp)
case expectWarning && resp == nil:
	t.Fatal("expected response (containing warning) when switching from enabled to disabled")
case expectWarning && len(resp.Warnings) == 0:
	t.Fatal("expected warning when switching from enabled to disabled")
}
```

Reply:
you can do logical ops in a go switch?? awesome! I'd googled quickly and didn't see examples